### PR TITLE
Added (rough draft of) weak form to manual / fixed strong form

### PR DESCRIPTION
**Pull request by rrgrove6**
Please check through weak form and make sure you agree.  Note that I wrote it as it appears in the code and it may still need more or better explanation!

Also, changing the strong form so that T is no longer on the right hand side.  I instead replaced it with a "T hat" and then added a description of what "T hat" is.

### CODE REVIEW

File: doc/manual/manual.tex [hunk 1]
```diff
@@ -449,14 +449,15 @@ \subsection{Basic equations}
...
+  +\alpha \hat{T} \left( \mathbf u \cdot \nabla p \right)
   \notag
   \\
   &\quad
-  + \rho T \Delta S \frac{\partial X}{\partial p} \mathbf u\cdot\nabla p
+  + \rho \hat{T} \Delta S \frac{\partial X}{\partial p} \mathbf u\cdot\nabla p
   & \quad & \textrm{in $\Omega$}.
   \notag
 \end{align}
+where $\hat{T}$ is an extrapolated temperature of the closest previous time steps.
```
Comment (tjhei) [inline doc/manual/manual.tex]: remove "closest"

File: doc/manual/manual.tex [hunk 2]
```diff
@@ -449,14 +449,15 @@ \subsection{Basic equations}
   \left(\varepsilon(\mathbf u) - \frac{1}{3}(\nabla \cdot \mathbf u)\mathbf 1\right)
   \\
   &\quad
-  +\alpha T \left( \mathbf u \cdot \nabla p \right)
+  +\alpha \hat{T} \left( \mathbf u \cdot \nabla p \right)
```
Comment (bangerth) [inline doc/manual/manual.tex]: Here and three lines below, it should remain to be `T`: the form discussed here is simply the continuous differential equation. There is no time stepping yet at this point, and consequently no need to approximate T in any way.

**Comment by bangerth** [comment]
I think this block of text is in the wrong place: it's in a section where we discuss the origin PDE. There are no time steps yet at this point, and consequently whether something appears on the right or the left side doesn't actually make a difference.

File: doc/manual/manual.tex [hunk 3]
```diff
@@ -466,6 +467,20 @@ \subsection{Basic equations}
...
 
+Starting with (\ref{eq:temperature-reformulated}) and multiplying on the left by a test function $\phi$ and integrating over the domain yields the weak form:
+\begin{align}
+  \label{eq:weak_form}
+  \left(\rho C_p - \rho T \Delta S \frac{\partial X}{\partial T}\right) 
+  \left( c \left( \phi, T \right) - \mathbf u \Delta t \left( \nabla \phi, T \right) 
+  \right) + k \Delta t \left( \nabla \phi , \nabla T \right)
+  &=
+  \left( \phi , \gamma \right) 
+  +  \left( \phi , q \right)
```
Comment (tjhei) [inline doc/manual/manual.tex]: one of the RHS terms should be multiplied by dt

File: doc/manual/manual.tex [hunk 4]
```diff
@@ -466,6 +467,20 @@ \subsection{Basic equations}
 purposes than just tracking chemical compositions. We will discuss this
 equation in more detail in Section~\ref{sec:compositional}.
 
+Starting with (\ref{eq:temperature-reformulated}) and multiplying on the left by a test function $\phi$ and integrating over the domain yields the weak form:
+\begin{align}
+  \label{eq:weak_form}
+  \left(\rho C_p - \rho T \Delta S \frac{\partial X}{\partial T}\right) 
+  \left( c \left( \phi, T \right) - \mathbf u \Delta t \left( \nabla \phi, T \right)
```
Comment (tjhei) [inline doc/manual/manual.tex]: u needs to be inside the integral, as does k.

Comment (tjhei) [inline doc/manual/manual.tex]: the "\rho T" should be "\rho \hat{T}"

**Comment by rrgrove6** [comment]
Wolfgang, was it the whole weak form you were refering to being in the wrong spot, or just the stuff about the lagging temperature?

**Comment by bangerth** [comment]
It's both. The statement about the weak form does not make very much sense in this place since it's simply showing that one has to multiply the equation with a test function and integrate by parts where relevant. That's mechanical. The interest in stating the weak form is in the context of the space and time discretization, because that involves lagging certain terms but not others, and adding spatial stabilization terms.

**Comment by bangerth** [comment]
So, in other words, the text would only be of interest in a section where we talk about the discretization, and in that case it should discuss the fully discrete form of the weak equation.

**Comment by tjhei** [comment]
@bangerth : We don't have a section for discretization in the manual (we should, though). His formula is after time discretization. Space discretization would just add a bunch of "_h" everywhere, right? (and specify different ansatz spaces).
Where would a "discretization section" go? 2.1.4? 2.12?

**Comment by bangerth** [comment]
I think 2.12 would be better, if we do want to add such a section. Though in that case I think stating just one formula would be the wrong approach. We should either strive for a self-contained discussion (probably several pages worth of text) or just reference one of the papers.

**Comment by rrgrove6** [comment]
Is there a happy medium where we could reference a paper for the fine details but maybe do a page or so of explanation that leads to the weak form?  That way people can still see the weak form and get a taste of where it comes from in the manual, and if they are more curious, they can look up the fine details?

**Comment by bangerth** [comment]
That would be ok with me, but maybe still put it into a section of its own (2.12). Please reference the KHB12 paper for more details. But if you do this, you need to identify which time step each appearance of the temperature is from, i.e., separate `T^n, T^{n-1}, T^{n-2}`, and `\hat T`.

**Comment by rrgrove6** [comment]
@tjhei shall we do this?

**Comment by tjhei** [comment]
In favor, yes.

**Pull request closed without merging**